enrollment.js

```diff
@@ -74,27 +75,70 @@ addInputHandler('enr_reg_start', function(input){ //input is first entry of nid
-    }
     else{
         state.vars.reg_nid = input;
-        sayText(msgs('enr_nid_confirm', {}, lang));
-        promptDigits('enr_nid_confirm', {'submitOnHash' : false, 'maxDigits' : max_digits_for_nid, 'timeout' : timeout_length});
+        var splash_menu = populate_menu('enr_nid_client_confirmation', lang, 300);
+        var current_menu = msgs('enr_nid_client_confirmation', {'$ENR_NID_CONFIRM' : input, '$ENR_CONFIRMATION_MENU' : splash_menu}, lang);
+        state.vars.current_menu_str = current_menu;
+        sayText(current_menu);
+        promptDigits('enr_nid_client_confirmation', {'submitOnHash' : false, 'maxDigits' : max_digits_for_input,'timeout' : timeout_length});
+
     }
-    get_time();
 });
 
+// Checking confirmation from the user
+addInputHandler('enr_nid_client_confirmation', function(input){
+    state.vars.current_step = 'enr_nid_client_confirmation';
+    input = String(input.replace(/\D/g,''));
+    var selection = get_menu_option(input, state.vars.current_step);
+    
+    if(input == 99){
+        sayText(msgs('exit', {}, lang));
+        stopRules();
+        return null;
+    }
+    else if(selection == null){
+        sayText(msgs('invalid_input', {}, lang));
+        promptDigits('invalid_input', {'submitOnHash' : false, 'maxDigits' : max_digits_for_input,'timeout' : timeout_length});
+    }
+    else{
+        var current_menu = msgs(selection, {}, lang);
+        state.vars.current_menu_str = current_menu;
+        
+
+        // If the user does not confirm
+        if(selection == 'enr_reg_start'){
+            sayText(current_menu);
+            promptDigits(selection, {'submitOnHash' : false, 'maxDigits' : max_digits_for_nid, 'timeout' : timeout_length});
+        }
+        //If the user confirms
+        else{
+            var is_already_reg = require('./lib/enr-check-dup-nid');
+            if(is_already_reg(state.vars.reg_nid , an_pool)){
+                var get_client_by_nid = require('./lib/dpm-get-client-by-nid');
+                var client = get_client_by_nid(state.vars.reg_nid , an_pool);
+                var enr_msg = msgs('enr_reg_complete', {'$ACCOUNT_NUMBER' : client.account_number, '$NAME' : client.name1 + ' ' + client.name2}, lang)
+                sayText(enr_msg);
+                var enr_msg_sms = msgs('enr_reg_complete_sms', {'$ACCOUNT_NUMBER' : client.account_number, '$NAME' : client.name1 + ' ' + client.name2}, lang);
+                var messager = require('./lib/enr-messager');
+                messager(contact.phone_number, enr_msg_sms);
+                if(state.vars.reg_pn){
+                    messager(state.vars.reg_pn, enr_msg_sms);
+                }
+                promptDigits('enr_continue', {'submitOnHash' : false, 'maxDigits' : max_digits_for_input,'timeout' : timeout_length});
+            }
+            // start registration process by asking them to enter their id again
+            else{
```

Comment:
Add comment details explicitly letting people know what current_menu and selection are